src/hotspot/share/gc/parallel/spaceCounters.hpp

```diff
@@ -39,6 +39,7 @@ class SpaceCounters: public CHeapObj<mtGC> {
  private:
   PerfVariable*      _capacity;
   PerfVariable*      _used;
+  volatile size_t    _last_used_in_bytes;
```

Comment:
Why is an additional field required here? Can't `_used` fulfill the purpose?

Reply:
The _used counters only seem to be updated by GCs.  The take_sample function reports "current" data.

Reply:
Using the perf-counter to carry the periodic samples doesn't work. During
construction of the perf-counter, the sampler is called to provide the initial
value for the counter. Since the counter is not yet fully constructed and
accessible to the sampling function (_used is not yet initialized), this crashes.

A workaround for that problem would be to add constructors that take both a
sampler and an initial value. That's kind of messy, and doesn't seem better
than the proposed solution at hand.

Reply:
Thank you for the clarification.